### PR TITLE
Updated AuSearch with Scarletts implementation

### PR DESCRIPTION
**Pull request by bee-san**
Please test the speed of this :-)

### CODE REVIEW

File: ciphey/basemods/Searchers/ausearch.py [hunk 1]
```diff
@@ -100,11 +100,130 @@ def get_path(self):
...
+@dataclass
+class AusearchResult:
+    weight: float
+    index: int
 
 def convert_edge_info(info: CrackInfo):
-    return cipheycore.ausearch_edge(
-        info.success_likelihood, info.success_runtime, info.failure_runtime
-    )
+    return 1
```
Comment (bee-san) [inline ciphey/basemods/Searchers/ausearch.py:123]: .

File: ciphey/basemods/Searchers/ausearch.py [hunk 2]
```diff
@@ -100,11 +100,130 @@ def get_path(self):
...
+
+# Equivalent to CPP ausearch_minimise
+def minimise_edges(edges) -> AusearchResult: 
+    weight = minimise_edges_impl(edges)
+    index = len(edges) # TODO: Is this ever different? C++ code seems to just count the number of elements...but in a really cursed way
```
Comment (bee-san) [inline ciphey/basemods/Searchers/ausearch.py:128]: .

File: ciphey/basemods/Searchers/ausearch.py [hunk 3]
```diff
@@ -186,7 +305,7 @@ def expand_crackers(self, node: Node) -> None:
         for i in self.get_crackers_for(type(res)):
             inst = self._config()(i)
             additional_work.append(
-                Edge(source=node, route=inst, info=convert_edge_info(inst.getInfo(res)))
+                Edge(source=node, route=inst)
```
Comment (bee-san) [inline ciphey/basemods/Searchers/ausearch.py:308]: .

File: ciphey/basemods/Searchers/ausearch.py [hunk 4]
```diff
@@ -259,15 +377,13 @@ def search(self, ctext: Any) -> Optional[SearchResult]:
                     #     chunk += self.work.get_work_chunk()
                     #     infos = [i.info for i in chunk]
 
-                    logger.trace(f"{len(infos)} remaining on this level")
-                    step_res = cipheycore.ausearch_minimise(infos)
-                    edge: Edge = chunk.pop(step_res.index)
+                    step_res = 0.1
+                    edge: Edge = chunk.pop(0)
```
Comment (bee-san) [inline ciphey/basemods/Searchers/ausearch.py:381]: .

**Review by SkeletalDemise — APPROVED** [review]

**Pull request merged**